AsyncDisplayKit/ASCollectionView.mm

```diff
@@ -882,20 +900,26 @@ - (CGSize)collectionView:(UICollectionView *)collectionView layout:(UICollection
 
 - (UICollectionReusableView *)collectionView:(UICollectionView *)collectionView viewForSupplementaryElementOfKind:(NSString *)kind atIndexPath:(NSIndexPath *)indexPath
 {
+  if ([_registeredSupplementaryKinds containsObject:kind] == NO) {
+    [self registerSupplementaryNodeOfKind:kind];
+  }
+  
   UICollectionReusableView *view = nil;
   ASCellNode *node = [_dataController supplementaryNodeOfKind:kind atIndexPath:indexPath];
 
-  if (node.shouldUseUIKitCell && _asyncDataSourceFlags.interop) {
+  BOOL shouldDequeueExternally = _asyncDataSourceFlags.interopViewForSupplementaryElement && (_asyncDataSourceFlags.interopAlwaysDequeue || node.shouldUseUIKitCell);
+  if (shouldDequeueExternally) {
     view = [(id<ASCollectionDataSourceInterop>)_asyncDataSource collectionView:collectionView viewForSupplementaryElementOfKind:kind atIndexPath:indexPath];
   } else {
-    ASDisplayNodeAssert(node != nil, @"Supplementary node should exist.  Kind = %@, indexPath = %@, collectionDataSource = %@", kind, indexPath, self);
-    if ([_registeredSupplementaryKinds containsObject:kind] == NO) {
-      [self registerSupplementaryNodeOfKind:kind];
-    }
     view = [self dequeueReusableSupplementaryViewOfKind:kind withReuseIdentifier:kReuseIdentifier forIndexPath:indexPath];
-    if (node) {
-      [_rangeController configureContentView:view forCellNode:node];
-    }
+  }
+
+  if (!node.shouldUseUIKitCell) {
```

Comment:
Totally stylistic, but might suggest using == NO in these cases for more explicitness / slightly improved readability